lib/Views/terriaStore.ts

```diff
@@ -0,0 +1,39 @@
+import { makeObservable, observable, action } from "mobx";
+import type Terria from "terriajs/lib/Models/Terria";
+import type ViewState from "terriajs/lib/ReactViewModels/ViewState";
+
+class TerriaStore {
+  terria: Terria | undefined = undefined;
+  viewState: ViewState | undefined = undefined;
+  status: "loading" | "ready" = "loading";
+
+  constructor() {
+    makeObservable(this, {
+      terria: observable,
+      viewState: observable,
+      status: observable,
+      setReady: action
+    });
+
+    this.init();
+  }
+
+  async init() {
+    //@ts-expect-error: need to convert to TS
```

Comment:
Does https://github.com/TerriaJS/terriajs/pull/7417 remove the need for this `ts-expect-error`?